deepinv/physics/generator/inpainting.py

```diff
@@ -474,15 +508,22 @@ def __init__(
             rng=rng,
         )
 
-    def batch_step(self, input_mask: torch.Tensor = None) -> dict:
+    def batch_step(
+        self, input_mask: torch.Tensor = None, img_size: Optional[tuple] = None
+    ) -> dict:
         if len(self.img_size) != 4:
-            raise ValueError("img_size must be of shape (C, T, H, W)")
+            raise ValueError("Default img_size must be of shape (C, T, H, W)")
 
-        if not isinstance(input_mask, torch.Tensor) or input_mask.numel() <= 1:
-            input_mask = torch.ones(self.img_size, device=self.device)
+        if input_mask is not None and input_mask.shape != self.img_size:
+            raise ValueError("input_mask must be same shape as default img_size")
```

Comment:
```suggestion
        if input_mask is not None:
           if input_mask.shape != self.img_size and img_size is None or img_size!=input_mask.shape:
            raise ValueError(f"input_mask must be same shape as default img_size. Got input_mask.shape {input_mask.shape:2e} and img_size {img_size:2e}.")
```
shouldnt there be a potential check on whether to use img_size or self.img_size?